src/app/core/params.ts

```diff
@@ -3,27 +3,22 @@ import Parameter from "./parameter";
  * Params class
  */
 export default class Params {
+    private unique: boolean;
+    private heritable: boolean;
+    parameters: any;
+    
+
     /**
      * Default constructor create a Params object.
      * @param {} values Values of the params
      * @param {Boolean} unique Boolean if it's unique
      * @param {} heritable Boolean if it's heritable
      * @param {} rawparameters ?
      */
-    constructor(values, unique, heritable, rawparameters) {
+    constructor(values: any, unique: boolean, heritable: boolean) { 
+        //rp must be optional for function further down. If no rp provided and values == null, what happens?
         this.unique = unique;
         this.heritable = heritable;
-        if (values != null) {
-            this.parameters = this.__sanitizeValues(values);
```

Comment:
`this.parameters` needs to be set. But without running `__sanitizeValues()`. So implement the logic for how the parameters property is being set and ignore all the other cleanup/validation pieces here.

Reply:
Looking at the way 'sanitizeValues' functioned, it appears 'values' only contains key strings, while the actual value/key pairings are found in either 'heritable' or 'unique'. That differs from how I now have these properties defined.